sktime/forecasting/model_selection/_split.py

```diff
@@ -281,6 +282,46 @@ def _check_cutoffs_fh_y(
         raise TypeError("Unsupported type of `cutoffs` and `fh`")
 
 
+def _check_freq_time_index(
+    y: pd.Index, window_length: Optional[ACCEPTED_WINDOW_LENGTH_TYPES]
+) -> Tuple[pd.Index, Union[pd.DateOffset, ACCEPTED_WINDOW_LENGTH_TYPES]]:
+    """Check that combination of inputs is compatible.
+
+    Parameters
+    ----------
+    y : pd.Index
+        Index of time series
+    window_length : int or timedelta or pd.DateOffset
+        Length of the window.
+
+    Returns
+    -------
+    y : pd.Index
+        Index of time series with inferred frequency.
+    window_length : pd.DateOffset
+        Length of the window with the correct format.
+    """
+    warn_msg = (
+        "The frequency of the time index is not set. "
+        "The frequency is inferred and set to {}."
+    )
+
+    error_msg = (
+        "Could not infer the frequency of the time index. "
+        "Please make sure window_length is the same datatype as y."
+    )
+
+    if isinstance(y, (pd.DatetimeIndex, pd.PeriodIndex)) and is_int(window_length):
+        if y.freq is None:
+            y.freq = pd.infer_freq(y)
+            if y.freq is None:  # failed to infer freq
+                raise ValueError(error_msg)
```

Comment:
oh yeah! but then just wondering how do then splitter splits irregular time series in this case? the window length still needs the freq information to get the wondow, no?

Reply:
well, maybe it should then break, or maybe we just use the iloc index (rather than freq).

The second sounds better, as it avoids any freq related issues.

Either way, we shouldn't restrict it on the level of the general class imo.

Reply:
Just to clarify so that my understanding inline with yours, so an irregular time series you are referring to refers to a time series that does not have any frequency information, right? For example, a datetime series with a 'day' frequency but may have gaps in the series where some dates in the index are not present at all (This is different from a date index with NaN values)

If that's the case then, the current implementation of `SingleWindowSplitter` asfaik cant deal with it because of the reliance in freq to change the window_length `int` to `DateOffset`. However, this drawback only happens when window_length is given as an `int` but not if window_length were provided  as a `DateOffset` in the first place. So I think when it breaks (when the freq cant is none) it is better to provide a message to pintpoint user to change window_length format. I think this suggestion will have minimal changes than having to change to use iloc index.

So for other general class, we can switch off the TypeError message when it is not needed?

Reply:
> Just to clarify so that my understanding inline with yours, so an irregular time series you are referring to refers to a time series that does not have any frequency information, right?

Almost - a time series that does not have the freq information, and that also cannot have it inferred.

> with gaps in the series where some dates if it were a day datetime in the index are not present at all

Yes

> I think this suggestion will have minimal changes than having to change to use iloc index.

Would the suggestion to use iloc result in different splits, in any case where it currently does not break?

If not, that is the minimal change imo

Reply:
Hi everyone, i have been taking a look at the PR. Given the case where the series is irregular, should we throw a warning in the logs and say we are using whatever consecutive data points we have or should we throw an error and say that there is data missing? In my opinion we could add an argument to the splitter, where in case some consecutive data points are missing we give the option to still use the sequence we get (even if it is incomplete). But the default maybe should be to throw an error. Let me know what you guys think!

Reply:
Also lets be careful here, i think you intend `window_length` to be an `int` or `timedelta`. The attribute `inferred_freq` is a string, so we this product of `y.inferred_freq* window_length`will return you a string.

Reply:
> Also lets be careful here, i think you intend window_length to be an int or timedelta. The attribute inferred_freq is a string, so this product of will make window_length a string.

not quite. the current implementation is rather as follow
```python
y.freq = y.inferred_freq 
window_length = y.freq * window_length
```
the y.inferred_freq string is converted to a `dateoffsets` format by `.freq` attribute so it wont make the `window_length` a string.